gprofiler/profilers/java.py

```diff
@@ -642,6 +648,10 @@ def _is_profiling_supported(self, process: Process) -> bool:
         return True
 
     def _profile_process(self, process: Process) -> Optional[StackToSampleCount]:
+        if not JavaProfiler._should_profile:
```

Comment:
You can also access it via `self` (if attr is not found on `self`, it looks up the attr on the class)